### ISSUE TITLE
Question : Metrics for reverse-proxy

### ISSUE THREAD

**Issue by aldevar** [hairyhenderson/caddyprom#8]
Hi, 
Here is my caddy configuration in json format
```json
{
    "storage":{
            "module": "consul"
    },
    "apps": {
        "http": {
            "servers": {
                "srv0": {
                    "listen": [
                        ":443"
                    ],
                    "routes": [
                        {
                            "handle": [
                                {
                                    "handler": "reverse_proxy",
                                    "upstreams": [
                                        {
                                            "dial": "10.0.0.1:80"
                                        },
                                        {
                                            "dial": "10.0.0.2:80"
                                        }
                                    ],
                                    "health_checks": {
                                        "active": {
                                            "path": "/",
                                            "port": 80,
                                            "interval": "0.5s",
                                            "timeout": "2s",
                                            "expect_status": 200
                                         }}
                                }
                            ],
                            "match": [
                                {
                                    "host": ["redacted.mycompany.com"]
                                }]
                        },
                    {
                            "handle": [
                                {
                                    "handler": "prometheus"
                                }
                            ],
                            "match": [
                                {
                                    "host": ["localhost"]
                                }
                            ]
                        }

                    ]
                }
            }
        }
    }
}
```

prometheux loads correctly and I'm able to view metrics. But metrics seems to be shown only when requesting localhost. If I generate some traffic on redacted.mycompany.com, the localhost:9180/metrics page does not change. No metric is added.

My question is, does this module works with reverse-proxy handler?

**Comment by hairyhenderson** [comment]
@aldevar it's likely an ordering thing - try placing the `prometheus` handler first.

**Comment by aldevar** [comment]
With the modified configuration file, I still have the same behavior.
```
{
    "storage":{
            "module": "consul"
    },
    "apps": {
        "http": {
            "servers": {
                "srv0": {
                    "listen": [
                        ":443"
                    ],
                    "routes": [{
                            "handle": [
                                {
                                    "handler": "prometheus"
                                }
                            ],
                            "match": [
                                {
                                    "host": ["localhost"]
                                }
                            ]
                        },
                        {
                            "handle": [
                                {
                                    "handler": "reverse_proxy",
                                    "upstreams": [
                                        {
                                            "dial": "10.0.0.1:80"
                                        },
                                        {
                                            "dial": "10.0.0.2:80"
                                        }
                                    ],
                                    "health_checks": {
                                        "active": {
                                            "path": "/",
                                            "port": 80,
                                            "interval": "0.5s",
                                            "timeout": "2s",
                                            "expect_status": 200
                                         }}
                                }
                            ],
                            "match": [
                                {
                                    "host": ["redacted.mycompany.com"]
                                }]
                        }
                    ]
                }
            }
        }
    }
}
```

I don't understand how I can make it work.
Once I succeed to have a working configuration, I'll push a PR with updated documentation.

**Comment by hairyhenderson** [comment]
@aldevar thanks, that'd be appreciated!

**Comment by aldevar** [comment]
Here is a working config
```javascript
{
    "apps": {
        "http": {
            "servers": {
                "srv0": {
                    "listen": [
                        ":443"
                    ],
                    "routes": [
                        {
                            "handle": [
                                {
                                    "handler": "subroute",
                                    "routes": [
                                        {
                                            "handle": [
                                                {
                                                    "handler": "prometheus"
                                                },
                                                {
                                                    "handler": "reverse_proxy",
                                                    "health_checks": {
                                                        "active": {
                                                            "expect_status": 200,
                                                            "interval": "0.5s",
                                                            "path": "/",
                                                            "port": 80,
                                                            "timeout": "2s"
                                                        }
                                                    },
                                                    "upstreams": [
                                                        {
                                                            "dial": "10.0.0.1:80"
                                                        },
                                                        {
                                                            "dial": "10.0.0.2:80"
                                                        }
                                                    ]
                                                }
                                            ]
                                        }
                                    ]
                                }
                            ],
                            "match": [
                                {
                                    "host": [
                                        "redacted.mycompany.com"
                                    ]
                                }
                            ],
                            "terminal": true
                        }
                    ]
                }
            }
        }
    },
    "storage": {
        "module": "consul"
    }
}
```

I'll push a PR with updated documentation when I have some time. Hopefully this evening

**Comment by hairyhenderson** [comment]
Thanks for doing this @aldevar! I've merged #11.
